internal/cli/user/profile_meta.go

```diff
@@ -0,0 +1,40 @@
+package user
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+// ProfileMeta contains the name and full filepath of a profile
+type ProfileMeta struct {
+	Name     string
+	Filepath string
+}
+
+// Profiles returns a list of each profile meta containing name and filepath
+func Profiles() ([]ProfileMeta, error) {
+	dir, dirErr := HomeDir()
+	if dirErr != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	dirEntryList, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	profileMetas := make([]ProfileMeta, 0, len(dirEntryList))
+	for _, v := range dirEntryList {
+		if !strings.Contains(v.Name(), profileType) {
```

Comment:
obviously wouldn't be able to use the `profileType = "yaml"` const if we moved to the `local` package.  lazily I'd say duplicating that is really fine, but we also already export `HostingAssetCacheDir` from this `user` package to re-use in the `local` package, so exporting `ProfileType` probably isn't the worst thing either